### DISCUSSION TITLE
Compression Streams should be automatic when elements have a decompress attribute.

### DISCUSSION THREAD

**Discussion by andrewhodel** [mdn/mdn-community#283] [Content]
https://developer.mozilla.org/en-US/docs/Web/API/Compression_Streams_API

While generating HTML from memory, parts of the stored HTML are not combined in memory and need to be compressed to save space in memory.

This would be nice.

```
<body>
<div decompress="zlib">zlib data</div>
</body>
```

Nicer than.

```
<script>
document.onload(function() {
    compressionStreamDecompressContents(document.getElementById('decompress_this_one'), 'zlib');
});
</script>

<body>
<div id="decompress_this_one">zlib data</div>
</body>
```

**Comment by Rumyra** [comment]
Hi @andrewhodel - thanks for opening a discussion.

Are you referring to the way the API works? If so that's outside of MDNs remit (we merely document the specs, not write them ourselves).

**Comment by andrewhodel** [comment]
That's why I wrote it in general, not as a feature request.



Andrew

**Reply by Rumyra** [reply]
Thanks @andrewhodel - the feedback is better given to the spec authors - see https://github.com/WICG/compression

**Comment by andrewhodel** [comment]
Thanks @Rumyra 

@stwrt please read this.

**Comment by bsmth** [comment]
Thanks for starting the discussion. I'm closing this now due to inactivity, but if you think that's a mistake, feel free to reply. You're also welcome to join any of our [communication channels](https://developer.mozilla.org/en-US/docs/MDN/Community/Communication_channels) if you'd prefer to chat with us more about it in sync. 

Thank you :)
